fetch upstream

80avin opened the pull request.

mpi3d committed:
Create FUNDING.yml

mpi3d committed:
Improved wireshark plugin

mpi3d committed:
Released driver 5110 and 5117

mpi3d committed:
Fixed tshark bug

mpi3d committed:
Renamed payload to hmac in check_firmware

mpi3d committed:
Renamed a field for better understanding

mpi3d committed:
Improved PSK stuff

mpi3d committed:
Added driver for 55b4

mpi3d committed:
Added command get_iap_version

mpi3d committed:
Improved null char handle for firmware_version

mpi3d committed:
Improved disection for "Get IAP Version"

mpi3d committed:
Changed psk address to psk flags

mpi3d committed:
Removed some unnecessary comments

mpi3d committed:
Removed default parameters to be more device specific

mpi3d committed:
Fix for 55b4

mpi3d committed:
Fix for 55b4

mpi3d committed:
Fix for 55b4

mpi3d committed:
Removed return before firmware flash

mpi3d committed:
Fix for 55b4

mpi3d committed:
Added IAP check for 55b4

mpi3d committed:
Removed printing "Abort"

mpi3d committed:
Removed return before driver

mpi3d committed:
Added reset sensor copy

mpi3d committed:
Security update

mpi3d committed:
Working on 55b driver

mpi3d committed:
Added second OTP

mpi3d committed:
Working on OTP

Commit:
wip psk

mpi3d committed:
Merge pull request #1 from 80avin/538d

wip psk

mpi3d committed:
Formated 538d stuff

mpi3d committed:
Fixed psk_r commands on 538d

mpi3d committed:
Nano fix

mpi3d committed:
Preparing driver for 538d

mpi3d committed:
Added firmware + psk cmd on 538 driver

mpi3d committed:
Released firmware GF5298_GM168SEC_APP_13016 for 538d

mpi3d committed:
Fix for 538d

mpi3d committed:
Added more flexibility for errors

mpi3d committed:
Added support to handle error for check firmware

mpi3d committed:
Improved code errors

mpi3d committed:
Fixed app data for 538d

mpi3d committed:
Fixed write_firmware cmd for 538d

mpi3d committed:
Added print before return

mpi3d committed:
Final commit before 538d test

mpi3d committed:
Fixed wait_disconnect command for 538d

mpi3d committed:
Added flashing part for 538d

mpi3d committed:
Test

mpi3d committed:
Final test for 538d

mpi3d committed:
Fixs for 55b4

mpi3d committed:
Fix for 55b4

mpi3d committed:
Fix for 55b4

mpi3d committed:
Increased read length

mpi3d committed:
Added pgm in .gitignore for more privacy

mpi3d committed:
Fix for 55b4

mpi3d committed:
Fix for 55b4 again

mpi3d committed:
Fix

mpi3d committed:
Fix again

mpi3d committed:
Fix for 55b4

mpi3d committed:
Added get image support for 55b4

mpi3d committed:
Preparing to wait finger for capturing image
(on 55b4)

mpi3d committed:
Renamed safe drivers

mpi3d committed:
Removed psk dissection

mpi3d committed:
Added POV image check command

mpi3d committed:
Increased timeout

mpi3d committed:
Added unknown class for dissection

mpi3d committed:
Added support for other protocols than USB

mpi3d committed:
Added mcu switch to sleep mode command

mpi3d committed:
Improved code readability

mpi3d committed:
Removed R after preset read and write PSK

mpi3d committed:
Removed MCU before sleep command

mpi3d committed:
Removed constant value

mpi3d committed:
Added number in dissection

mpi3d committed:
Added waiting finger support for 55b4

mpi3d committed:
Changed 55b4 driver to safe

mpi3d committed:
Removed security

mpi3d committed:
Added get image for 538d

mpi3d committed:
Fix for 538d

mpi3d committed:
Fix for 538d

mpi3d committed:
Added success to mcu erase app

mpi3d committed:
Changed read length for 538d

mpi3d committed:
Added 521 driver

mpi3d committed:
Fix for 521d

mpi3d committed:
Fix for 521d

mpi3d committed:
Fix for 538d

mpi3d committed:
New fix for 538d

mpi3d committed:
Fix for 538d

mpi3d committed:
Fix for 538d

mpi3d committed:
Added submodule firmware

mpi3d commented:
I removed all firmware binaries from all commits. I now use a submodule for firmware. This permit better security about the firmware proprietor.

You should consider rebasing your repo on mine to remove all firmware binaries too.

mpi3d committed:
Added *.bin

mpi3d committed:
Update FUNDING.yml

mpi3d committed:
Update FUNDING.yml

mpi3d committed:
Update FUNDING.yml

mpi3d committed:
Update FUNDING.yml

mpi3d committed:
Update FUNDING.yml

mpi3d committed:
Update FUNDING.yml

mpi3d committed:
Added mcu_switch_to_sleep_mode cmd

mpi3d committed:
Fixed cmd query_mcu_state for 538d

mpi3d committed:
Fix for query_mcu_state

mpi3d committed:
Fixed cmd mcu_switch_to_fdt_down for 538d

mpi3d committed:
Added wait finger support for 538d

mpi3d committed:
Added firmware for 55a

mpi3d committed:
Added README in firmware

mpi3d commented:
You should take a look at https://gitlab.freedesktop.org/mpi3d/goodix-firmware/-/blob/main/README.md

mpi3d commented:
And also at https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository

mpi3d committed:
Set sleep_time in mcu_erase_app for 55b4

mpi3d committed:
Added 55a driver

mpi3d committed:
Fixed valid firmwares

mpi3d committed:
Fix for 55a4

mpi3d committed:
Fix for 55a4

mpi3d committed:
Removed tries for reading psk

mpi3d committed:
Using global function for write psk

mpi3d committed:
Added cmd set_drv_state for 521d

mpi3d committed:
Renamed cmd mcu get pov image

mpi3d committed:
Added cmd mcu_get_pov_image

mpi3d committed:
Added cmd set_pov_config

mpi3d committed:
Split long lines

mpi3d committed:
Fixed command mcu_erase_app for 538d

mpi3d committed:
Added driver for 521d

mpi3d committed:
Setting resolution for 521d

mpi3d committed:
Removed "_nonsafe" for 521d

mpi3d committed:
Added second OTP

mpi3d committed:
Improved code format

mpi3d committed:
Removed comment for 511

mpi3d committed:
Removed tries for write firmware

mpi3d committed:
Removed tries for update firmware

mpi3d committed:
Hard coded firmware path

mpi3d committed:
Restricted errors on firmware update

mpi3d committed:
Added tools

mpi3d committed:
Grouped driver parts

mpi3d committed:
Sorted imports

mpi3d committed:
Split long lines

mpi3d committed:
Sorted functions

mpi3d committed:
Added init device function

mpi3d committed:
Added OTP check

mpi3d committed:
Added 538c driver

mpi3d committed:
Added 5381 driver

mpi3d committed:
Fix for 5381

mpi3d committed:
Fix for 5381

mpi3d committed:
Last commit of the morning

mpi3d committed:
Minor change

mpi3d committed:
Changed 538d as safe

mpi3d committed:
Removed driver for 5381

mpi3d committed:
Added 5385 driver

mpi3d committed:
Removed device 5385

mpi3d committed:
Added wireshark disector without wrapper

mpi3d committed:
Worked on OTP

mpi3d committed:
Added OTP check

mpi3d committed:
Added firmware for 5117

mpi3d committed:
Improved disection

mpi3d committed:
Splited driver 511

mpi3d committed:
Renamed NAV command

mpi3d committed:
Fix for 5117

mpi3d committed:
Fix for 5117

mpi3d committed:
Fix for 5117

mpi3d committed:
Fix for 5117

mpi3d committed:
Fix for 5110

mpi3d committed:
Improved disection

mpi3d committed:
Added driver for 55a4

mpi3d committed:
Removed 538c

mpi3d committed:
Renamed driver names

mpi3d committed:
Updated firmware for 51x0

mpi3d committed:
Removed firmware GF3208_RTSEC_APP_10052

mpi3d committed:
Added 532d device

twemyss committed:
Use new USBError object

twemyss committed:
Update 51x7 driver to collect images with new config code

twemyss committed:
Tidy up 51x7 driver

twemyss committed:
Use newer libusb with specific errors

twemyss committed:
pycodestyle fixes

rootd committed:
Merge pull request #3 from twemyss/master

Capture image on Goodix 5117

mpi3d committed:
Added png file to gitignore

atar-axis committed:
v3.7

atar-axis committed:
Update README.md

atar-axis committed:
Update README.md

atar-axis committed:
Merge pull request #1 from atar-axis/patch-1

added some text to README and python v3.7

mpi3d committed:
Merge pull request #4 from atar-axis/master

added some txt to README, lowered python version necessary

EliaGeretto committed:
Update wrapless protocol dissector using 5395

mpi3d committed:
Merge pull request #6 from EliaGeretto/wrapless

Update wrapless protocol dissector using 5395

EliaGeretto committed:
Add initial support for 53x5 sensors

mpi3d committed:
Merge pull request #7 from EliaGeretto/wrapless

Add initial support for 53x5 sensors

EliaGeretto committed:
wrapless: Fix return type annotations

EliaGeretto committed:
wrapless: Add support for GTLS handshake

EliaGeretto committed:
Improve wrapless dissector for image replies

EliaGeretto committed:
wrapless: Add support to read chip ID and OTP

EliaGeretto committed:
wrapless: Fix magic checksum check

EliaGeretto committed:
wrapless: Switch from mbedtls to PyCryptodome

EliaGeretto committed:
wrapless: Add GTLS decryption function

EliaGeretto committed:
wrapless: Add test code to get image from sensor

mpi3d committed:
Merge pull request #8 from EliaGeretto/wrapless

Add 53x5 GTLS code and image capture script

mpi3d committed:
Update README.md

egormanga committed:
Added SPI protocol support.

There's been a whole lot of research behind this.

mpi3d committed:
Merge pull request #9 from egormanga/spidev

Added SPI protocol support

EliaGeretto committed:
53x5: Fully implement device init

EliaGeretto committed:
53x5: Implement finger detection and finger image reading

mpi3d committed:
Merge pull request #10 from EliaGeretto/wrapless

53x5: Fully implement init, finger detection and fingerprint reading

Timarrr committed:
made write_pgm write actual pgm

Timarrr committed:
fixed addfp script for 5110 sensor

Timarrr committed:
fixed another thing in script

mpi3d committed:
Merge pull request #17 from Timarrr/master

made write_pgm write actual pgm

EliaGeretto committed:
53x5: Add deactivation commands

EliaGeretto committed:
53x5: Relax firmware requirements

EliaGeretto committed:
53x5: Fix type warnings

mpi3d committed:
Merge pull request #19 from EliaGeretto/deactivate

53x5: Fix deactivation and firmware requirements

mpi3d committed:
Remove .vscode

mpi3d committed:
Improve code readability

mpi3d committed:
Merge branch 'master' of github.com:mpi3d/goodix-fp-dump

mpi3d committed:
Clean the branch

mpi3d committed:
Format add_fp_5110.sh

mpi3d committed:
Update Wireshark readme

mpi3d committed:
Update Python to 3.10

mpi3d committed:
Update readme

mpi3d committed:
Update logs readme

mpi3d committed:
Fix Crypto components import

juriSacchetta committed:
Fix Crypto components import

mpi3d committed:
Merge pull request #21 from juriSacchetta/master

Fix Crypto components import

mpi3d committed:
Update Bitcoin address

EliaGeretto committed:
5395: Add flasher to fix broken sensors

mpi3d committed:
Merge pull request #24 from EliaGeretto/flasher_53x5

5395: Add flasher to fix broken sensors

EliaGeretto committed:
flasher_53x5: Check only first 2 chars of chip vendor

EliaGeretto committed:
53x5: Fix firmware update loop condition

mpi3d committed:
Merge pull request #25 from EliaGeretto/flasher_53x5

53x5: Fixes for flasher

hmvp committed:
Add requirements.txt to make it easier to install required depencies

hmvp committed:
Add run file for 5385 (confirmed it working)

mpi3d committed:
Merge pull request #26 from hmvp/5385

Add support for 5385 model

EliaGeretto committed:
53x5: Add flash dumper

EliaGeretto committed:
Update wrapless dissector for FLASH messages

EliaGeretto committed:
wrapless: Check firmware write result

mpi3d committed:
Merge pull request #28 from EliaGeretto/dumper_53x5

53x5: Add dumper script

0x00002a committed:
fix: flashing 5110 failing

mpi3d committed:
Merge pull request #29 from 0x00002a/0x2a/fix/5110-flash

fix: flashing 5110 failing

EliaGeretto committed:
wrapless: Add comments for cmd0: 0xa

EliaGeretto committed:
wrapless: Add comments for cmd0: 0xc

EliaGeretto committed:
wrapless: Fix formatting for cmd0: 0xf

EliaGeretto committed:
wrapless: Implement write_option_byte

mpi3d committed:
Update submodule url

mpi3d committed:
Update submodule

mpi3d committed:
Update LICENSE

EliaGeretto committed:
dumper_53x5: Add code to dump OTP and USB PID buffers

EliaGeretto committed:
wrapless: Add USB PID commands to dissector

EliaGeretto committed:
wrapless: Add various notes to dissector

EliaGeretto committed:
53x5: Add script to print default config

mpi3d committed:
Merge pull request #30 from EliaGeretto/firmware-reverse

wrapless: Add comments on protocol

EliaGeretto committed:
53x5: Add dump scripts for 5385 and 5335

hadron43 committed:
Remove $ for easy copy paste

Signed-off-by: Harsh Kumar <hadron43@yahoo.com>

rootd committed:
Update README hyperlinks to goodix-fp-linux-dev

Signed-off-by: Alexander Meiler <rootd@users.noreply.github.com>

anarsoul committed:
Add trivial preprocessor

It just subtracts background from raw image and improves contrast
by doing histogram equalization (with fixed white)

It seems to be enough to make sigfm happy with 5395 scanner

Signed-off-by: Vasily Khoruzhick <anarsoul@gmail.com>

mpi3d committed:
Merge pull request #33 from anarsoul/master

Add trivial preprocessor

anarsoul committed:
proprocessor: drop hardcoded black and white levels

It appears to work pretty well with clamped black/white levels
if the border is properly discarded. Otherwise black level is
estimated incorrectly. For white level we need to discard pixels
that are not covered.

Signed-off-by: Vasily Khoruzhick <anarsoul@gmail.com>

anarsoul committed:
preprocessor: add TODO for implementing coverage estimation

Current implementation is very primitive and won't work well for
all the cases

Signed-off-by: Vasily Khoruzhick <anarsoul@gmail.com>

mpi3d committed:
Merge pull request #34 from anarsoul/master

preprocessor: drop hardcoded black and white levels

anarsoul committed:
preprocessor: crop input images, use thresholding and mean filters

- Dealing with black border adds a lot of corner cases, so just crop
the image.
- Use thresholding filter for detecting finger coverage. That also
allows us to just drop bin 0 (black) and bin 255 (white) while doing
histogram equalization and gets rid of white level clamping
- Use mean filter to improve image sharpness
- Refactor the code, move image processing into their own functions
instead of abusing main
- Write out intermediate images for debugging purposes

Signed-off-by: Vasily Khoruzhick <anarsoul@gmail.com>

coredoesdev committed:
Functional 5503 driver

mpi3d committed:
Merge pull request #37 from c0repwn3r/master

Add a functional 5503 driver

mpi3d committed:
update firmware submodule
